File: Back-End/backend/Models/Holyplace.js
import mongoose from 'mongoose';

const HolyplaceFields = {

    name :{
        type:String,
        default:null,
        required:true,
        
    },
    location :{
        type:String,
        default:null,
        required:true
    },
    history:{
        type:String,
        default:null,
        required:true
    },
    image:{
        type:String,
        default:null,
    }



}


const HolyPlaceSchema = Schema(HolyplaceFields);

const HolyPlaceModel = model('holyplaces' , HolyPlaceSchema);

export default HolyPlaceModel;

File: Front-End/frontend/lib/DataProvider/Holyplace/Holyplace.dart
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'package:frontend/Models/models.dart';

class HolyPlaceDataProvovider {
  static Future<dynamic> getAllHolyplaces() async {
    List holyplaces = [];

    try {
      final httpresponse = await http.get(
        Uri.parse('http://localhost:3500/api/getallholyplaces'),
      );

      if (httpresponse.statusCode == 200) {
        var finalresult = jsonDecode(httpresponse.body);
        holyplaces = finalresult['holyplaces'];
      }
    } catch (e) {}

    return holyplaces;
  }

  static Future<dynamic> createHolyPlace(HolyplaceModel holyplaceModel) async {
    var responsemessage = "";

    try {
      print("on dataprovider");

      final response = await http
          .post(Uri.parse('http://localhost:3500/api/createHolyPlace'), body: {
        'createdby': holyplaceModel.createtor_id.toString(),
        'name': holyplaceModel.nameofholyplace.toString(),
        'location': holyplaceModel.location.toString(),
        'history': holyplaceModel.history.toString(),
        'image': holyplaceModel.imageUrl.toString()
      });
      print(holyplaceModel.createtor_id);
      responsemessage = jsonDecode(response.body).toString();
      print("response: $responsemessage");
    } catch (e) {
      print(e.toString());
    }
    return responsemessage;
  }
}
